hydra-node/src/Hydra/Logging/Monitoring.hs

```diff
@@ -96,6 +98,8 @@ monitor transactionsMap metricsMap = \case
     tick "hydra_head_requested_tx"
   (Node LogicOutcome{outcome = Continue{stateChanges}}) -> do
     forM_ stateChanges $ \case
+      PeerConnected{} -> gauge Gauge.inc "hydra_head_peers_connected"
+      PeerDisconnected{} -> gauge Gauge.dec "hydra_head_peers_connected"
```

Comment:
We currently do *NOT* guarantee to have a `PeerDisconnected` for each peer if we disconnect from the network. Instead, the `NetworkDisconnected` should be interpreted as "nobody connected" and to have a correct gauge, we should set it to 0 in that case.